src/main/kotlin/com/demonwav/mcdev/util/McBytecodeUtil.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+@file:JvmName("McBytecodeUtil")
+package com.demonwav.mcdev.util
+
+import com.intellij.psi.PsiArrayType
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiClassType
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.PsiPrimitiveType
+import com.intellij.psi.PsiType
+import com.intellij.psi.util.TypeConversionUtil
+
+private const val INTERNAL_CONSTRUCTOR_NAME = "<init>"
+
+// Type
+
+internal val PsiPrimitiveType.internalName: Char
+    get() = when (this) {
+        PsiType.BYTE -> 'B'
+        PsiType.CHAR -> 'C'
+        PsiType.DOUBLE -> 'D'
+        PsiType.FLOAT -> 'F'
+        PsiType.INT -> 'I'
+        PsiType.LONG -> 'J'
+        PsiType.SHORT -> 'S'
+        PsiType.BOOLEAN -> 'Z'
+        PsiType.VOID -> 'V'
+        else -> throw IllegalArgumentException("Unsupported primitive type: $this")
+    }
+
+private fun PsiClassType.erasure() = TypeConversionUtil.erasure(this) as PsiClassType
+
+internal val PsiClassType.internalName
+    get() = erasure().resolve()!!.internalName
```

Comment:
I'll see if I can fix it up, then, before I pull. In the instances where these are used, `resolve` shouldn't ever return null, but I promise you it can and will. I understand your reasoning for why you did it this way, but I can't pull this PR like that. I don't think it will be very difficult to handle these situations (or at the very least just continuing when the errors are thrown for now) so I don't think I'll have too much of an issue fixing these up.